### ISSUE TITLE
Skip feedstock CI builds that don't produce binaries

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/conda-smithy#234]
I was thinking of other optimizations we could get. This seems to be a big one to me. Often there are PRs where we are adding a maintainer or re-rendering that won't actually ship a binary. As the CIs already ran on the PR, I see no point in doing it a second time for no package. The idea here would be to fast finish CIs by recognizing no binary will ship and thus not building at all.

**Comment by pelson** [comment]
This is kind of the same as the capability that conda-build-all gives you. It inspects the destination channel to determine if it needs to build or not. I have been reluctant to use that in the feedstocks though, as it means we aren't using the vanilla conda-build tools.

**Comment by jakirkham** [comment]
Yeah, I'm just noting the AppVeyor (and sometimes Travis) queues are getting pretty backed up. When I look through the queue, I sometimes find unactionable builds like re-renders or metadata updates. It seems like an easy win to speed these up a bit.

**Comment by jakirkham** [comment]
> I have been reluctant to use that in the feedstocks though, as it means we aren't using the vanilla conda-build tools.

Could you please explain this rationale a bit more? What concerns do you think using `conda-build-all` or other process to determine whether the build should run unveils?

**Comment by pelson** [comment]
> What concerns do you think using conda-build-all or other processes to determine whether the build should run unveils?

FWIW one problem that we have managed to get away from with individual feedstocks is the concept that you need to bump the version or the build id in order to assert that any changes you made don't break the repo. The healthier approach, IMO, is to *test* the recipe by building, not to use metadata to ascertain whether this recipe has already been built and is therefore a no-op.

That said, there are other advantages to using conda-build-all - one such advantage is the ability to do a matrix of builds within the same CI matrix (like we do with CircleCI anyway). This could be useful to speed up trivial packages that have a big matrix and that are swamping CI because they are spending all of their time creating the *build* environment, and not doing the building itself.

**Comment by jakirkham** [comment]
To be more clear about what I'm asking, we are spending two build cycles to generate a package we don't deliver to anyone. The first in a PR and the second after the merge. It seems like there is a lot of savings (especially for long running feedstocks) if we can acknowledge the PR build was sufficient and not occupy X more hours of CI time for no real benefit.

**Comment by jakirkham** [comment]
It's worth noting that `conda-build` has a `--skip-existing` flag that enables this behavior. So it would be easy for us to add if we would like to have it.

Credits to @isuruf for [reminding]( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/94#issuecomment-409344750 ) us about this.
